### PR TITLE
fix: add enableClientSideSearch

### LINKED ISSUE #5444: [Bug] Table search not working with onSearchTextChanged
## Description

For the table widget, if the `onSearchTextChanged` action is defined the actual search functionality is not working anymore. The `onSearchTextChanged` action itself is executed, but the search (filter) on the table is not applied.

### Steps to reproduce the behaviour:

1. Create a table widget with some data.
2. Set `onSearchTextChanged` to some action (e.g. save the search value in the store or show a message)
3. Enter a search term. The search filter is not applied, but the action is executed.
4. Remove the `onSearchTextChanged` action.
5. Enter a search term. The search filter is now applied.

### Important Details

- Version: self-hosted, v1.5.5
- OS: Windows 10
- Browser: Chrome
- Environment: production

### PR DESCRIPTION
**Pull request by Tooluloope**
Fixes #5444 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: Feat/Disable-Client-Side-Search 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.05 **(0)** | 36.92 **(0)** | 33.92 **(0.01)** | 55.57 **(0)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/derived.js | 58.77 **(0.12)** | 59.39 **(-0.37)** | 46.51 **(1.27)** | 58.76 **(0.12)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/widget/derived.js [hunk 1]
```diff
@@ -420,17 +420,22 @@ export default {
       },
     };
 
-    const searchKey =
-      props.searchText && !props.onSearchTextChanged
-        ? props.searchText.toLowerCase()
-        : "";
+    const searchKey = () => {
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidget/widget/derived.js:423]: Lets change this name to getSearchKey

File: app/client/src/widgets/TableWidget/widget/derived.js [hunk 2]
```diff
@@ -420,17 +420,22 @@ export default {
...
+      if (props.searchText && !props.onSearchTextChanged) {
+        return props.searchText.toLowerCase();
+      }
+      if (props.searchText && props.enableClientSideSearch) {
+        return props.searchText.toLowerCase();
+      }
+      return "";
+    };
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidget/widget/derived.js:431]: Could be refactored to 
<pre>
if (props.searchText && (!props.onSearchTextChanged || props.enableClientSideSearch)) {
    return props.searchText.toLowerCase();
}
return "";
</pre>

**Comment by YogeshJayaseelan** [comment]
@Tooluloope Tested this PR and came up with the below issue 

Issue 1 - Default search is not working 

https://www.loom.com/share/c8d9c82267f34190b038f29719419a47

**Review by sbalaji1192 — APPROVED** [review]
LGTM

**Comment by Tooluloope** [comment]
@YogeshJayaseelan Fixed

**Comment by vivekverma2312** [comment]
@Tooluloope Tested Deploy preview and issue is fixed.

**Comment by Nikhil-Nandagopal** [comment]
@Tooluloope are we giving people an option to turn this on / off or is it implicit? Making it implicit will break it for existing apps.

**Comment by Tooluloope** [comment]
@Nikhil-Nandagopal We are giving them the option to turn it on / off

**Comment by Nikhil-Nandagopal** [comment]
@Tooluloope thanks for the confirmation! :)

**Pull request merged**